dotnet/src/Microsoft.Agents.AI.DevUI/Conversations/Models/ConversationItem.cs

```diff
@@ -0,0 +1,210 @@
+            {
+                var contentObj = new Dictionary<string, object>
+                {
+                    ["type"] = "text",
+                    ["text"] = textContent.Text ?? string.Empty
+                };
+
+                // Include additional properties if present
+                if (textContent.AdditionalProperties?.Count > 0)
+                {
+                    foreach (var kvp in textContent.AdditionalProperties)
+                    {
+                        if (kvp.Value is not null)
+                        {
+                            contentObj[kvp.Key] = kvp.Value;
+                        }
+                    }
+                }
+
+                contentArray.Add(contentObj);
+            }
+            else
+            {
+                // For non-text content, serialize using our context
+                // This works in both AOT and non-AOT scenarios
+                var json = JsonSerializer.Serialize(content, ConversationsJsonUtilities.DefaultOptions.GetTypeInfo(typeof(AIContent)));
+                var element = JsonDocument.Parse(json).RootElement;
+                contentArray.Add(element);
+            }
+        }
+
+        // If no content items were added, create a default empty text content
+        if (contentArray.Count == 0 && !string.IsNullOrEmpty(chatMessage.Text))
+        {
+            contentArray.Add(new Dictionary<string, object>
+            {
+                ["type"] = "text",
+                ["text"] = chatMessage.Text
+            });
+        }
+
+        // Serialize using ConversationsJsonContext for AOT compatibility
+        var contentJson = JsonSerializer.Serialize(contentArray, ConversationsJsonContext.Default.ListDictionaryStringObject);
+        var contentElement = JsonDocument.Parse(contentJson).RootElement;
+
+        return new ConversationItem
+        {
+            Id = id ?? $"msg_{Guid.NewGuid():N}",
+            Type = "message",
+            ConversationId = conversationId,
+            CreatedAt = DateTimeOffset.UtcNow.ToUnixTimeSeconds(),
+            Role = chatMessage.Role,
+            Content = contentElement,
+            Metadata = metadata
+        };
+    }
+
+    /// <summary>
+    /// Creates a Message with text content.
+    /// </summary>
+    public static ConversationItem CreateTextMessage(ChatRole role, string text, string? conversationId = null, string? id = null, Dictionary<string, string>? metadata = null)
+    {
+        var contentArray = new[]
+        {
+            new Dictionary<string, object>
+            {
+                ["type"] = "text",
+                ["text"] = text
+            }
+        };
+
+        // Serialize using ConversationsJsonContext for AOT compatibility
+        var contentJson = JsonSerializer.Serialize(contentArray, ConversationsJsonContext.Default.DictionaryStringObjectArray);
+        var contentElement = JsonDocument.Parse(contentJson).RootElement;
```

Comment:
https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-10/#json and search for JsonDocument.Parse :) We created a small pit of failure, unfortunately.